lib/proxy/helpers.rb

```diff
@@ -64,13 +64,24 @@ def parse_json_body
     json_data
   end
 
+  def dns_resolv(*args)
+      resolv = Resolv::DNS.new(*args)
+      resolv.timeouts = Proxy::SETTINGS.dns_resolv_timeouts
+      ::Proxy::LoggingResolv.new(resolv)
+  end
+
+  def resolv(*args)
+      ::Proxy::LoggingResolv.new(Resolv.new(*args))
+  end
+
   # reverse lookup an IP address while verifying it via forward resolv
   def remote_fqdn(forward_verify=true)
     ip = request.env['REMOTE_ADDR']
     log_halt 403, 'could not get remote address from environment' if ip.empty?
 
     begin
-      dns = Resolv.new
+      # TODO - shall we actually make this dns_resolv?
+      dns = resolv
       fqdn = dns.getname(ip)
```

Comment:
Here in this remote_fqdn method I realized that we don't actually do DNS resolve but generic resolve, this will use any OS provider (kerberos, NSS, DNS). Do we actually want to keep this behavior or shell I change this to strictly DNS only?

Reply:
Ping @ekohl

Reply:
Sorry, I missed this. I'm leaning to the current resolv. I think the value of using NSS is actually pretty big and something we should be preferring unless we really need DNS.